src/main/java/org/apache/sling/feature/impl/OrderedDictionary.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.sling.feature.impl;
+
+import java.util.*;
+
+/**
+ * A dictionary implementation with predictable iteration order.
+ *
+ * Actually this class is a simple adapter from the Dictionary interface
+ * to a synchronized LinkedHashMap
+ *
+ * @param <K>
+ * @param <V>
+ */
+public class OrderedDictionary<K, V> extends Dictionary<K, V> implements Map<K, V> {
```

Comment:
Do you need the Map implementation as well? In `Configuration.java` it is only used as a `Dictionary` implementation.

Reply:
@rombert Yes, that is true, in `Configuration.java` it is only used as `Dictionary`, but I wanted to maintain backwards compatibility with all the other code that unknowingly used the implementation details of `Hashtable`. For example I've seen a lot of code doing things like `object instance of Map` and those conditions would fail.

Reply:
My personal preference would be to keep it simple in the beginning and add the extra baggage only when it's needed.